### PR TITLE
Podman CLI --add-host with multiple host for a single IP

### PR DESCRIPTION
**Pull request by jerome59**
Hello, 

Fixes https://github.com/containers/podman/issues/23770

I didn't update any tests because I don't know which one I should update and how to update them without breaking them.
It would be nice if someone creates the tests for me or give me some explanation on where and how I can create a test case. 

#### Does this PR introduce a user-facing change?

```release-note
--add-host option now supports multiple hostnames

```

### CODE REVIEW

File: vendor/github.com/containers/common/libnetwork/etchosts/hosts.go [hunk 1]
```diff
@@ -252,7 +253,8 @@ func parseExtraHosts(extraHosts []string, hostContainersInternalIP string) (Host
 			}
 			ip = hostContainersInternalIP
 		}
-		e := HostEntry{IP: ip, Names: []string{values[0]}}
```
Comment (Luap99) [inline vendor/github.com/containers/common/libnetwork/etchosts/hosts.go:255]: you cannot change vendor code directly, this must be first submitted as PR in https://github.com/containers/common then we can update the dependency here

Comment (jerome59) [reply vendor/github.com/containers/common/libnetwork/etchosts/hosts.go:255]: Yes I've seen that, that's why I just opened this PR : https://github.com/containers/common/pull/2139 

So on this repository, I should just remove my changes from hosts.go ?

Comment (jerome59) [reply vendor/github.com/containers/common/libnetwork/etchosts/hosts.go:255]: As the PR has been merged, how do we proceed to include a version of the common library into this code ? 
I've seen that it is defined there https://github.com/containers/podman/blob/main/go.mod#L16 and https://github.com/containers/podman/blob/main/vendor/modules.txt#L174 

I guess someone from the core team has to create a new tag 0.60.3 and includes the commit in it and then we can refer to it in the previous files

Comment (Luap99) [reply vendor/github.com/containers/common/libnetwork/etchosts/hosts.go:255]: I let you know once we updated it in podman, likely sometime next week. We generally vendor based of the main branch during development so we do not need a new tag there.

Comment (jerome59) [reply vendor/github.com/containers/common/libnetwork/etchosts/hosts.go:255]: Hello @Luap99, any news on the previous point ?

Comment (Luap99) [reply vendor/github.com/containers/common/libnetwork/etchosts/hosts.go:255]: Sorry I was on PTO so I forget to update you here. Yes, you can rebase the change is vendored now.

Comment (jerome59) [reply vendor/github.com/containers/common/libnetwork/etchosts/hosts.go:255]: I guess I'm missing something here :

I don't see any new release of the common lib : https://github.com/containers/common/releases 
And the dependency seems to still be defined in 0.60.1 : https://github.com/containers/podman/blob/main/go.mod#L16

Comment (Luap99) [reply vendor/github.com/containers/common/libnetwork/etchosts/hosts.go:255]: We vendor by commit during dev time on main, we only cut releases of the libs when we cut a new podman.
For example https://github.com/containers/podman/commit/b9fe4097ddb98385af85f9417e2601aac313833c.
You can check in the vendor/ directory your changes are in there.

**Comment by jerome59** [comment]
I've rebased + added some tests so I guess the PR is now ready for review

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**